src/main/java/org/elasticsearch/common/util/concurrent/PrioritizedEsThreadPoolExecutor.java

```diff
@@ -133,10 +129,13 @@ public Pending(Object task, Priority priority, long insertionOrder, boolean exec
         }
     }
 
-    static class TieBreakingPrioritizedRunnable extends PrioritizedRunnable {
+    private class TieBreakingPrioritizedRunnable extends PrioritizedRunnable {
```

Comment:
I think we can add a simple unittest that submits a task with a huge timeout that just checks if the task is still in the queue after it has run? 

We can do this for both queues the executor as well as the timeout scheduler? They both expose getQueue and we can iterate there I guess?